server/org.eclipse.emfcloud.ecore.glsp/src/main/java/org/eclipse/emfcloud/ecore/glsp/model/EcoreModelServerAccess.java

```diff
@@ -97,16 +99,78 @@ private EPackage getEPackage(EcoreModelState modelState) {
-		return this.add(modelState, getEPackage(modelState), EcorePackage.Literals.EPACKAGE__ECLASSIFIERS,
-				newEClassifier);
+		return this.add(modelState, getEPackage(modelState), EcorePackage.eINSTANCE.getEPackage_EClassifiers(), newEClassifier);
+	}
+	
+	public boolean addSuperType(EcoreModelState modelState, EClassifier superType, EClassifier parent) {
+		return this.add(modelState, parent, EcorePackage.eINSTANCE.getEClass_ESuperTypes(), superType);
+	}
+	
+	public boolean addEReference(EcoreModelState modelState, EReference newEReference, EClassifier parent) {
+		return this.add(modelState, parent, EcorePackage.eINSTANCE.getEClass_EStructuralFeatures(), newEReference);
+	}
+	
+	public boolean addEAttribute(EcoreModelState modelState, EAttribute newEAttribute, EClassifier parent) {
+		return this.add(modelState, parent, EcorePackage.eINSTANCE.getEClass_EStructuralFeatures(), newEAttribute);
+	}
+	
+	public boolean addEEnumLiteral(EcoreModelState modelState, EEnumLiteral newEEnumLiteral, EEnum parent) {
+		return this.add(modelState, parent, EcorePackage.eINSTANCE.getEEnum_ELiterals(), newEEnumLiteral);
+	}
+	
+	public boolean setOpposite(EcoreModelState modelState, EReference eReference, EReference opposite) {
+		return this.set(modelState, eReference, EcorePackage.eINSTANCE.getEReference_EOpposite(), opposite);
+	}
+	
+	public boolean setInstanceName(EcoreModelState modelState, EClassifier eClassifier, String name) {
+		return this.set(modelState, eClassifier, EcorePackage.eINSTANCE.getEClassifier_InstanceClassName(), name);
+	}
+	
+	public boolean setName(EcoreModelState modelState, EClassifier eClassifier, String name) {
+		return this.set(modelState, eClassifier, EcorePackage.eINSTANCE.getENamedElement_Name(), name);
+	}
+	
+	public boolean setAttributeName(EcoreModelState modelState, EAttribute eAttribute, String name) {
+		return this.set(modelState, eAttribute, EcorePackage.eINSTANCE.getENamedElement_Name(), name);
+	}
+	
+	public boolean setAttributeType(EcoreModelState modelState, EAttribute eAttribute, EClassifier type) {
+		return this.set(modelState, eAttribute, EcorePackage.eINSTANCE.getEAttribute_EAttributeType(), type);
+	}
+	
+	public boolean setLiteralName(EcoreModelState modelState, EEnumLiteral eEnumLiteral, String literal) {
+		return this.set(modelState, eEnumLiteral, EcorePackage.eINSTANCE.getENamedElement_Name(), literal);
+	}
+	
+	public boolean setAttributeType(EcoreModelState modelState, EAttribute eAttribute, EAttribute type) {
+		return this.set(modelState, eAttribute, EcorePackage.eINSTANCE.getEAttribute_EAttributeType(), type);
+	}
+	
+	public boolean setEdgeName(EcoreModelState modelState, EReference eReference, String name) {
+		return this.set(modelState, eReference, EcorePackage.eINSTANCE.getENamedElement_Name(), name);
+	}
+	
+	public boolean setLowerMultiplicity(EcoreModelState modelState, EReference eReference, int multiplicity) {
```

Comment:
This method can be generalized to accept any EStructuralFeature (i.e. either an EAttribute or an EReference). While the diagram doesn't show the EAttribute multiplicity yet, we'll probably add it in the future (Via the EAttribute text parser, similar to the EAttribute Type)